src/gui/qgsattributeform.cpp

```diff
@@ -223,14 +223,15 @@ void QgsAttributeForm::setMode( QgsAttributeForm::Mode mode )
   emit modeChanged( mMode );
 }
 
-void QgsAttributeForm::changeAttribute( const QString &field, const QVariant &value )
+void QgsAttributeForm::changeAttribute( const QString &field, const QVariant &value, const QString &toolTip )
 {
   Q_FOREACH ( QgsWidgetWrapper *ww, mWidgets )
   {
     QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( ww );
     if ( eww && eww->field().name() == field )
     {
       eww->setValue( value );
+      eww->widget()->setToolTip( toolTip );
```

Comment:
Mmm I'm not convinced by this kind of hidden logic. I think it would be strange to finally get a placeholder text after having explicitely set the content to empty.